force_bdss/mco/optimizer_engines/weighted_optimizer_engine.py

```diff
@@ -125,7 +142,19 @@ def _weighted_optimize(self, weights):
 
         weighted_score_func = partial(self.weighted_score, weights=weights)
 
-        return self._scipy_optimize(weighted_score_func)
+        # use default .optimize_function
+        optimal_point = self.optimize_function(
+            weighted_score_func,
+            self.initial_parameter_value,
+            self.parameter_bounds)
+        optimal_kpis = self._score(optimal_point)
+
```

Comment:
Here we call the ScipyOptimize.optimize_function(). If we create a child of WeightedOptimizerEngine that also inherits say NevergradOptimize as the *first* inheritance arg, it will call NevergradOptimize.optimize_function() instead.

Reply:
Alternative
```
class optimizer_a:
    def get_name(self):
        return 'a'
    
class optimizer_b:
    def get_name(self):
        return 'b'
    
class engine_j():
    
    def get_name(self):
        """ Dummy function (or leave out in the expectation...)"""
        return 'none'
    
    def get_optimizser_name(self):
        return 'optimizer = ' + self.get_name()

class engine_q(optimizer_b, engine_j):
    pass

print(engine_j().get_optimizser_name())

print(engine_q().get_optimizser_name())
```

Reply:
> Have you experimented with implementing an appropriate `NevergradOptimize.optimize_function` method? Is this an appropriate signature for both external libraries?

Yes

Reply:
We can dynamically create engine/optimizer mixes (in reference to a user selection).

```
myengine = type('MyEngine', (optimizer_b, engine_j), {})
print(myengine().get_optimizser_name())
```

Reply:
... the only thing with the dummy method, is that you still end up overriding it when you inherit an optimizer, so it really doesn't make any difference (to overiding an inheritance from another optimizer).